src/common/redis_module/ray_redis_module.cc

```diff
@@ -762,35 +762,45 @@ int ReplyWithTask(RedisModuleCtx *ctx,
     RedisModuleString *local_scheduler_id = NULL;
     RedisModuleString *execution_dependencies = NULL;
     RedisModuleString *task_spec = NULL;
+    RedisModuleString *spillback_count = NULL;
     RedisModule_HashGet(key, REDISMODULE_HASH_CFIELDS, "state", &state,
                         "local_scheduler_id", &local_scheduler_id,
                         "execution_dependencies", &execution_dependencies,
-                        "TaskSpec", &task_spec, NULL);
+                        "TaskSpec", &task_spec, "spillback_count",
+                        &spillback_count, NULL);
     if (state == NULL || local_scheduler_id == NULL ||
-        execution_dependencies == NULL || task_spec == NULL) {
+        execution_dependencies == NULL || task_spec == NULL ||
+        spillback_count == NULL) {
       /* We must have either all fields or no fields. */
       RedisModule_CloseKey(key);
       return RedisModule_ReplyWithError(
           ctx, "Missing fields in the task table entry");
     }
 
     long long state_integer;
-    if (RedisModule_StringToLongLong(state, &state_integer) != REDISMODULE_OK ||
-        state_integer < 0) {
+    long long spillback_count_value;
+    int state_status = RedisModule_StringToLongLong(state, &state_integer);
+    int spillback_count_status =
+        RedisModule_StringToLongLong(spillback_count, &spillback_count_value);
+    if (state_status != REDISMODULE_OK || state_integer < 0 ||
+        spillback_count_status != REDISMODULE_OK || spillback_count_value < 0) {
       RedisModule_CloseKey(key);
       RedisModule_FreeString(ctx, state);
       RedisModule_FreeString(ctx, local_scheduler_id);
       RedisModule_FreeString(ctx, execution_dependencies);
       RedisModule_FreeString(ctx, task_spec);
-      return RedisModule_ReplyWithError(ctx, "Found invalid scheduling state.");
+      RedisModule_FreeString(ctx, spillback_count);
+      return RedisModule_ReplyWithError(
+          ctx, "Found invalid scheduling state or spillback count.");
     }
 
     flatbuffers::FlatBufferBuilder fbb;
     auto message =
         CreateTaskReply(fbb, RedisStringToFlatbuf(fbb, task_id), state_integer,
                         RedisStringToFlatbuf(fbb, local_scheduler_id),
                         RedisStringToFlatbuf(fbb, execution_dependencies),
-                        RedisStringToFlatbuf(fbb, task_spec), updated);
+                        RedisStringToFlatbuf(fbb, task_spec),
+                        spillback_count_value, updated);
```

Comment:
This is the crux of it. The `updated` value was being passed in to the `TaskReply` for the `spillback_count` argument.